src/app/core/services/service-cache-interceptor.ts

```diff
@@ -0,0 +1,101 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpRequest,
+  HttpResponse,
+  HttpInterceptor,
+  HttpHandler
+} from '@angular/common/http';
+import { ServicesCache } from './services-cache';
+import { catchError, map, reduce, tap, first } from 'rxjs/operators';
+import { Observable, of } from 'rxjs';
+import { OperatorFunction } from 'rxjs';
+
+import { TagInfoService } from './tag-info.service';
+import { SelectizeOption } from '../../shared/interfaces/selectize-option.interface';
+import { environment } from 'src/environments/environment';
+
+// Influenced by the following!
+// -> https://fullstack-developer.academy/caching-http-requests-with-angular/
+// -> https://www.concretepage.com/angular/angular-caching-http-interceptor
+
+@Injectable()
+export class ServiceCacheInterceptor implements HttpInterceptor {
+  constructor(private cache: ServicesCache) { }
+
+  /**
+   * Intermediate step before http client sends a request usable to use cache when there
+   * and also cache responses
+   * @param req {HttpRequest<any>} request to interrogate regarding how it should be handled...
+   * @param next {HttpHandler} handler we use when we send a request and don not use cache
+   * @return {Observable<any>} Observable with response (be it from cache or endpoint)...
+   */
+  intercept(req: HttpRequest<any>, next: HttpHandler) {
+    if (!this.shouldCache(req)) {
+      return next.handle(req);
+    }
+
+    let requestObservable: Observable<any>;
+    const cached = this.cache.get(req);
+
+    if (cached !== null) {
```

Comment:
logic of saying (if I already requested this, then get from cached, otherwise, request)